src/status_im2/contexts/communities/discover/events.cljs

```diff
@@ -0,0 +1,45 @@
+(ns status-im2.contexts.communities.discover.events
+  (:require [clojure.set :as set]
+            [clojure.string :as string]
+            [taoensso.timbre :as log]
+            [utils.re-frame :as rf]))
+
+(defn camel->kebab
```

Comment:
or actually if its so simple function, probably we don't need to use external library, and better to use this one but move it to utils

Reply:
I've seen that library, but haven't fully explored the pros/cons of using for this. The current solution is pretty small and specific to communities (we have some special handling here, for predicates and public keys that the library isn't going to do for us).

Reply:
I don't see a reason to not use the library. It's a well known, tested, small library in the Clojure community, so implementing our own transformation functions is just a waste of time IMO. It's already in our `shadow-cljs.edn`.

>have you seen this (utils.collection/map-keys csk/->kebab-case-keyword data) in the code, so it seems it could be used 

@flexsurfer, this implementation works very well for converting the map keys, but it's not recursive. Which is also good in some occasions if we're sure we don't need to recursively convert the data structures. For a recursive solution, we just need to use `cske/transform-keys` combined with `csk/->kebab-case-keyword`, both provided by the library.

Every major Clojure codebase I worked with that had to interface with JSON APIs needed a recursive transformation function for converting keys.

For those places where performance is critical we might want to benchmark and do a more direct conversion. Maybe using `assoc` + `dissoc` could lead to faster results than `rename-keys` or `cske/transform-keys`, but I have no clue. But again, this would only be necessary if we can meaningfully measure a slow down.

Reply:
I think we should use the library at least to transform from camel -> kebab since it's already implemented and tested, reimplement the recursive key transformation seems a lot of work.

btw, why do we have keywords starting with a number?

> Converts keys that start with a digit (0-9) from keywords to strings (:0x0324 -> "0x0324")

AFAIK, this kind of keywords are supported because early Clojure wrongly allowed them to exist so they keep supporting them, but they are not recommended.

Reply:
@flexsurfer @ilmotta @ulisesmac Thanks for the feedback!

 I added some tests and renamed the functions, to give a better idea of what's involved here. Also, here's what the input data looks like: [wakuext_curatedCommunities](https://gist.github.com/erikseppanen/9e37fdbc2661be963a5380a43f005d25)
 1. JSON API: the data is not strictly JSON, although it does include camel->kebab conversion. For example, the keys are already keywords, and it performs additional operations that are not related to camel->kebab conversion (renaming predicates, stringifying number-prefixed keys, and recursion to transform nested maps). The camel->kebab conversion is just one line of code: `(string/lower-case (string/replace s #"([a-z])([A-Z])" "$1-$2"))`, which looks for a lowercase letter followed by an uppercase letter, puts a dash between them, then lowercases everything. This camel->kebab conversion is maybe 10% of the solution.
 2. Motivation: I wrote these to replace the existing `communities.core` `set/rename-keys` [functions](https://github.com/status-im/status-mobile/blob/develop/src/status_im/communities/core.cljs#L64). While working on this PR, I noticed that not only would would it require another `<-` function, but that there were already many camelcase words in the data that had been missed (not converted to kebab case). So I concluded that the current key-by-key approach leads to keys being overlooked.
 3. Specific to contract communities: The current key-by-key renaming approach is used many times in the codebase, but I can't testify to those conversions following the same rules as contract communities here.
 4. General utility function: Creating a general utility function would entail additional analysis of the conversion rules of all the other data structures that use the key-by-key renaming approach, well suited for a separate PR that can be prioritized appropriately.
 5. `camel-snake-kebab` library: I've used this library before, for transforming mongodb JSON documents. In that case, an approach was needed that would convert documents that were universally JSON, with little control over the the actual field names. It is important to note that the library does not presume the format it is converting FROM: it has to do the work of figuring out whether the input string is PascalCase, HTTP-Header-Case, or whatever before converting it to the desired case ([non-trivial code](https://github.com/clj-commons/camel-snake-kebab/blob/master/src/camel_snake_kebab/internals/string_separator.cljc#L42)). Our use-case here is to convert a single data structure once, at startup, whose fieldnames we can reliably predict because we define them. While I don't think we necessarily need a library for this, I went ahead and replaced the line of code I mentioned above with `(csk/->kebab-case-string s)`.